core/src/main/scala/org/apache/spark/ui/jobs/JobProgressListener.scala

```diff
@@ -93,6 +93,8 @@ class JobProgressListener(conf: SparkConf) extends SparkListener with Logging {
 
   val retainedStages = conf.getInt("spark.ui.retainedStages", SparkUI.DEFAULT_RETAINED_STAGES)
   val retainedJobs = conf.getInt("spark.ui.retainedJobs", SparkUI.DEFAULT_RETAINED_JOBS)
+  val retainedTasks = conf.getInt("spark.ui.retainedTasks", SparkUI.DEFAULT_RETAINED_TASKS)
```

Comment:
Ah, now I get where you where going with that, yeah if it wasn't for the map size issue that would have been good.